docs/build/guides/conventions/soroban-contract-init-template.mdx

```diff
@@ -0,0 +1,432 @@
+  );
+  //save contractId and alias to contracts Object
+  contractsObject[alias] = {
+    alias,
+    id: contractId,
+  };
+});
+
+//Create a json File to store the contract name as key and contract id and contract alias contained in an object as value
+const contractsJSONPath = "./contracts.json";
+const contractsJSONData = JSON.stringify(contractsObject);
+
+try {
+  writeFileSync(contractsJSONPath, contractsJSONData);
+  console.log(`JSON file has been saved to ${contractsJSONPath}`);
+} catch (err) {
+  console.error("Error writing JSON file: ", err);
+}
+
+//Generate Contract Typescript Bindings
+let filename, output_dir;
+files.forEach((file) => {
+  filename = path.basename(file).split(".")[0];
+  output_dir = `./packages/${filename}`;
+  run(
+    `stellar contract bindings typescript --output-dir ${output_dir} --network ${process.env.NETWORK} --contract-id ${contractsObject[alias]["id"]}  --wasm ${file} --overwrite`,
+  );
+});
+```
+
+## Modifying `package.json` file
+
+After saving the `initialize.js` script, let's modify our `package.json` file to create a new entry under scripts to run the code to work on the contract files.
+
+Let's also modify the file to create a new workspace in the `package.json` to point to the path where the the TypeScript bindings are generated
+
+```json
+"scripts": {
+    "setup": "node initialize.js",
+  },
+
+ "workspace": ["./packages/*"],
+```
+
+## Running the script
+
+Now that we have a complete script and a frontend, we should try to run the script. However, because the script is dependent on contract files we need to have a contract project ready, otherwise the script will fail and result in an error. To solve this issue we have to either copy the whole project contents and paste them to an existing Stellar contract project directory or upload the template file to a repo then create a new project and setting the 'f' parameter to point to the hosted repository of the template.
+
+### Setting up the project
+
+A community member has hosted a [public repository of the React template](https://github.com/CaptainPrinz/soroban-react-template) available for use. The following command implements the second option:
+
+```bash
+stellar contract init `hello_contract` -w hello -f https://github.com/CaptainPrinz/soroban-react-template.git
+```
+
+The snippet above initializes a new Stellar smart contracts project, `hello_contract`, that uses a prebuilt `hello_world` contract and adds the frontend template we deployed to a remote repository.
+
+### Setting the environment variables
+
+Our templates offer a great deal of customization via environment variables. These variables need to be set by the user by creating a `.env` file otherwise the process will fail as these variables will be undefined.
+
+### Starting the script
+
+On your terminal run the following code:
+
+```bash
+npm run setup
```

Comment:
Your template repository has a [git conflict saved](https://github.com/CaptainPrinz/soroban-react-template/blob/c2dbaf770005411c87aa2c9b1110aa202cd734fd/initialize.js#L30-L34) in the `initialize.js` file, so this step fails with an error.

Reply:
Running through the process again, two things I've come across:

```suggestion
npm install
cp .env.example .env # edit .env file as needed
npm run setup
```

First, let's be **overt** and say `npm install` needs to run. It's a bit "goes without saying," but I think it'll be common enough for someone going through the tutorial step-by-step to forget about, so let's just be extra clear.

Second, if the `.env.example` file isn't copied over, the initialization script will error out and complain about network and account configs missing.

Third, I'm getting a couple errors when running `npm run setup`. Here's all the output:

```bash
❯ npm run setup          

> hello_contract@0.0.0 setup
> node initialize.js

Configuring Network
Error executing command: 
 Command failed: stellar network add testnet --rpc-url http://soroban-testnet.stellar.org:443 --network-passphrase Test SDF Network ; September 2015
error: unexpected argument 'SDF' found

Usage: stellar network add [OPTIONS] --rpc-url <RPC_URL> --network-passphrase <NETWORK_PASSPHRASE> <NAME>

For more information, try '--help'.
/bin/sh: September: command not found

Here comes the output

Existing Files 
Here comes the output

[ 'target/wasm32-unknown-unknown/release/hello_world.wasm' ]
Deploying Contract hello_world
Here comes the output

JSON file has been saved to ./contracts.json
Error executing command: 
 Command failed: stellar contract bindings typescript --output-dir ./packages/hello_world --network testnet --contract-id CC2L3OKAGKIQZOGP57I76Z7PPVJWTYQ3S6D7SXVQTAMN63TBAFVITWFC
  --wasm target/wasm32-unknown-unknown/release/hello_world.wasm --overwrite
error: --output-dir already exists and you did not specify --overwrite: "./packages/hello_world"
/bin/sh: line 1: --wasm: command not found
```

In your `initialize.js` script, you may want to change [this part](https://github.com/CaptainPrinz/soroban-react-template/blob/2d642ed1aa7e6ad941dead572c1d5a0c43a462eb/initialize.js#L24) to something like this, so the network passphrase gets completely enclosed in quotes.

```js
run(`stellar network add ${process.env.NETWORK} --rpc-url ${process.env.RPC_URL} --network-passphrase "${process.env.NETWORK_PASSPHRASE}"`)
```

There seems to be a trail of other errors in the initialization script that need to be tracked down and figured out. Let me know if you need any help with it.

Reply:
Fixed